File: src/lib-common/pop.js
function pop(x,y)
{
 check(is_arr,x)
 
 if(is_undef(y))
  return pop(x,1)
 
 check(is_uint,y)

 const n=sub(x.length,y)
 
 if(same(y,1))
 {
  const r=back(x)
  
  remove(x,n,1)
  
  return r
 }
 
 remove(x,n,y)
}

File: src/lib-common/pop.cs
fn pop x:arr y
 if is_undef y
  ret pop x 1

 check is_uint y
 check lte y x.length

 let n sub x.length y

 if same y 1
  let r back x

  remove x n 1

  ret value r
 end

 remove x n y
end
